src/Command/Argument/Search/Schema.php

```diff
@@ -0,0 +1,191 @@
+ */
+
+namespace Predis\Command\Argument\Search;
+
+use Predis\Command\Argument\ArrayableArgument;
+
+class Schema implements ArrayableArgument
+{
+    /**
+     * @var array
+     */
+    private $arguments = ['SCHEMA'];
+
+    /**
+     * Adds text field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addTextField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('TEXT', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds tag field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addTagField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('TAG', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds numeric field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addNumericField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('NUMERIC', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds geo field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addGeoField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('GEO', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds vector field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addVectorField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('VECTOR', $fieldName, $alias, $sortable, $unf, $noIndex);
```

Comment:
`VECTOR` does not support `SORTABLE` and have more options, see [here](https://redis.io/docs/stack/search/reference/vectors/)

Reply:
Fixed